client/tinychain/math/operator.py

```diff
@@ -101,6 +101,99 @@ def backward(self):
         return Mul(self.args, Pow(self.subject, Sub(self.args, 1)))
 
 
+class Sin(Operator):
+    def __init__(self, subject):
+        Operator.__init__(self, subject, None)
```

Comment:
You should add a `Unary` operator type so you don't have to repeat this definition of `__init__` for every operator:
```python
class Unary(Operator):
  def __init__(self, subject):
    Operator.__init__(self, subject, None)

class Sin(Unary):
  # ...

# ...
```